include/NAS2D/MathUtils.h

```diff
@@ -29,5 +30,26 @@ namespace NAS2D
 
 	int divideUp(int to_divide, int divisor);
 
+	
+
+/**
+ * \fn T convertDomainToRange(T a, T b)
+ *
+ * Applies a linear conversion to map an input domain to an output range for built-in types.
+ *
+ * \param	value: The value to be converted from the input domain to the output range.
+ * \param	minDomain: Smallest value of the domain.
+ * \param	maxDomain: Largest value of the domain.
+ * \param	minRange: Smallest value of the range.
+ * \param	maxRange: Largest value of the range.
```

Comment:
These parameter names are perhaps slightly incorrect. This function actually works quite well outside of the min/max bounds, by simple linear extension. Instead, these parameters mark two corresponding fixed points, which sets the linear scale being used. They do not need to be at the extreme ends of the scale.

Typical values may be the extreme ends, so that's fine, and might be worth a comment. I would however like to document the function works outside of those bounds by linear extension.

Since naming is hard, I suppose I should at least attempt to provide a suggestion:
- `domainPoint1`: Typically the minimum value for domain
- `domainPoint2`: Typically the maximum value for domain
- `rangePoint1`: Typically the minimum value for range
- `rangePoint2`: Typically the maximum value for range
We should perhaps also specify that corresponding points are mapped to each other other, while other values are linearly interpolated.

----

Boundary case:
- I think it's fine if `rangePoint1` coincides with `rangePoint2`. In that case, all inputs produce the same output value.
- We should document what happens if `domainPoint1` coincides with `domainPoint2`. We could to return `rangePoint1` for all values, or return `rangePoint2` for all values, or we could `throw` an exception. Actually, We basically get the division by 0 case here, so probably throw an exception.

Reply:
Documented expected inputs to the function.